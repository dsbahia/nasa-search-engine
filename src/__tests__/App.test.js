import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  it("Renders Correctly", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Displays NASA Logo", () => {
    render(<App />);
    const logoElement = screen.getByAltText("Nasa Logo");
    expect(logoElement).toBeInTheDocument();
    expect(logoElement).toHaveAttribute("src", "https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg");
  });

  it("Displays Search Component", () => {
    render(<App />);
    const searchElement = screen.getByPlaceholderText("Enter keywords...");
    expect(searchElement).toBeInTheDocument();
  });
});
