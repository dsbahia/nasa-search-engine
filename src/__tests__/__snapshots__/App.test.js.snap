// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <img
      alt="Nasa Logo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div>
      <form>
        <input
          class="search"
          placeholder="Enter keywords..."
          type="text"
          value=""
        />
        <button
          class="search-button"
          type="submit"
        >
          Go!
        </button>
      </form>
    </div>
    <div
      class="search-results-container"
    >
      <div
        class="image-grid"
      />
    </div>
  </div>
</DocumentFragment>
`;
